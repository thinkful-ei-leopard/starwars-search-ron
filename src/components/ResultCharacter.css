.ResultCharacter__list {
    background-color: aquamarine;
}