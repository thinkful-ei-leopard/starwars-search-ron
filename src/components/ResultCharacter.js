import React from 'react';

export default function ResultCharacter(props) {

    // destructing the array. blank spaces are skipped values.
    //const [name, , , hair_color, , eye_color, birth_year, gender, homeworld, ...rest] = props;
    // console.log(rest);

    return (
        <li>
            <p>{props.name}</p>
            {/* <p>{name}
            {hair_color}
            {eye_color}
            {birth_year}
            {gender}
            {homeworld}
            </p> */}
        </li>
    );
}