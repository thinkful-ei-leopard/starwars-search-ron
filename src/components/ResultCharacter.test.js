
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ResultCharacter from './ResultCharacter';

describe('<ResultCharacter />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ResultCharacter />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<ResultCharacter />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});