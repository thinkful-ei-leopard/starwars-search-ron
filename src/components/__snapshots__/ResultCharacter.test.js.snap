// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultCharacter /> renders this UI as expected 1`] = `
<li
  className="ResultCharacter__list"
>
  <p
    className="result_name"
  >
    Name: undefined
  </p>
  <p
    className="result_name"
  />
  <p
    className="result_name"
  >
    Hair color: undefined, Eye color: undefined
  </p>
  <p
    className="result_name"
  >
    Birth year: undefined
  </p>
</li>
`;
